### PR TITLE
[posix-app] fix using GNU readline on macOS

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: src/posix/README.md [hunk 1]
```diff
@@ -122,3 +122,19 @@ make -f src/posix/Makefile-posix DAEMON=1
...
+
+Readline Support
+----------------
+
+libedit seems to be buggy supporting callback mode, so GNU readline is recommended.
+
+```bash
+# macOS requires setting these environments
+brew install readline
+export LDFLAGS="-L$(brew --prefix)/opt/readline/lib"
```
Comment (jwhui) [inline src/posix/README.md]: Can/should these be added to `configure.ac` or `src/posix/Makefile-posix`?

**Comment by bukepo** [comment]
@jwhui I updated according to your suggestions. Also, I'm wondering if we should use readline by default.

**Comment by jwhui** [comment]
> I'm wondering if we should use readline by default.

Are you suggestiong to effectively have `--with-readline=readline` even when no such parameter is specified on `./configure`? If so, I am fine with that.

**Comment by bukepo** [comment]
@jwhui updated in `src/posix/Makefile-posix'

**Pull request merged**